samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```
@@ -0,0 +1,181 @@
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/common/eventHeader.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover/alexaDiscoveryDiscoverDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover/alexaDiscoveryDiscoverDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover.Response/alexaDiscoveryDiscoverResponseEvent.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover.Response/alexaDiscoveryDiscoverResponseEventPayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/SetIndicator/notificationsSetIndicatorDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/SetIndicator/notificationsSetIndicatorDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirectivePayload.proto
+    ${CMAKE_CURRENT_SOURCE_DIR}/proto/custom_event.proto
+)
+
+# Populate list of expected .c files output from protobuf generation
+foreach(file ${PROTO_FILES})
+    get_filename_component(FILE_PATH ${file} DIRECTORY)
+    get_filename_component(FILE_BASE ${file} NAME_WE)
+    list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+    list(APPEND PROTO_BYPRODUCTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+endforeach()
+
+# Populate list of compilation commands for protoc compiler.
+foreach(file ${PROTO_FILES})
+    get_filename_component(FILE_PATH ${file} DIRECTORY)
+    get_filename_component(FILE_NAME ${file} NAME)
+    set(PROTOC_ARGS "${PROTOC_INC_DIRS}")
+    list(APPEND PROTOC_ARGS "--nanopb_out=${PROTO_COMPILATION_DIR}")
```

Seems the `-I.` is needed, but the list remove is not:
> `accessories.proto: File does not reside within any path specified using --proto_path (or -I).  You must specify a --proto_path which encompasses this file.`

but this will suffice:
```suggestion
    list(INSERT PROTOC_ARGS 0 "-I.")
    list(APPEND PROTOC_ARGS "--nanopb_out=${PROTO_COMPILATION_DIR}")
```

but you really don't need to use list commands here, this will work the same:
```suggestion
    set(PROTOC_ARGS "-I."
                    "${PROTOC_INC_DIRS}"
                    "--nanopb_out=${PROTO_COMPILATION_DIR}"
                    "${FILE_NAME}"
    )
```

---

Curiously the "-I." is not needed when using the apt-installed version of the proto compiler. Updating the PROTOC_ARGS variable to include this parameter.